streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -1104,11 +1104,12 @@ private void initializeTopology() {
         // initialize the task by initializing all its processor nodes in the topology
         log.trace("Initializing processor nodes of the topology");
         for (final ProcessorNode<?, ?, ?, ?> node : topology.processors()) {
-            processorContext.setCurrentNode(node);
+            final InitProcessorRecordContext initContext = new InitProcessorRecordContext(time);
+            updateProcessorContext(node, time.milliseconds(), initContext);
             try {
                 node.init(processorContext, processingExceptionHandler);
             } finally {
-                processorContext.setCurrentNode(null);
+                updateProcessorContext(null, RecordQueue.UNKNOWN, null);
```

Comment:
Might be better to use `ConsumerRecord.NO_TIMESTAMP` directly instead of `RecordQueue.UNKNOWN` ?